roxie/ccd/ccdprotocol.cpp

```diff
@@ -2160,6 +2161,15 @@ class RoxieSocketWorker : public ProtocolQueryWorker
                         VStringBuffer xml(" <wuid>%s</wuid>\n", statsWuid.str());
                         response.flushXML(xml, true);
                     }
+                    if (queryPT->getPropBool("@summaryStats", alwaysSendSummaryStats))
+                    {
+                        FlushingStringBuffer response(client, (protocolFlags & HPCC_PROTOCOL_BLOCKED), mlResponseFmt, (protocolFlags & HPCC_PROTOCOL_NATIVE_RAW), false, logctx);
+                        response.startDataset("SummaryStats", NULL, (unsigned) -1);
+                        VStringBuffer s(" COMPLETE: %s %s complete in %u msecs memory=%u Mb agentsreply=%u duplicatePackets=%u resentPackets=%u resultsize=%u continue=%d", queryName.get(), uid, elapsed, memused, agentsReplyLen, agentsDuplicates, agentsResends, bytesOut, continuationNeeded);
+                        logctx.getStats(s).newline();
+                        response.flushXML(s, true);
```

Comment:
Does this also support output in json format?   If so this might need to think about encoding/quoting the string?
From a quick search it looks ok, but will leave as a question.

Reply:
I don't know what if any json support there is for Roxie responses. There are other bits of code above that appear to assume that the data to be sent to flushXML is raw XML data though.